### PR TITLE
Added signing service spec + mock

### PR DESCRIPTION
**Pull request by pandermatt**
Added signing service spec and http mock

### CODE REVIEW

File: src/app/services/signing/http.mock.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { Observable } from 'rxjs';
+import { SigningResponse } from './signing-response.interface';
+
+@Injectable()
+export class HttpMock {
+  static getProviders() {
+    return {provide: Http, useClass: this};
+  }
+
+  lastMethod: 'post';
```
Comment (pandermatt) [inline src/app/services/signing/http.mock.ts]: Because we only use `post`

File: src/app/services/signing/http.mock.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        x_amz_algorithm: 'dfe',
+        x_amz_credential: 'saklde/eu-central-1/s3/aws4_request',
+        x_amz_expires: 2000,
+        x_amz_signature: 'qwerty',
+        x_amz_date: '20171212T',
+        utf8: '✓'
+      },
+      file_prefix: 'pre/fix',
+      file_url_path: '//example.com/o/upload-demo-testing'
+    });
```
Comment (lukaselmer) [inline src/app/services/signing/http.mock.ts]: imo this mock data does not belong in this general `HttpMock` - i would put it to the spec where it is needed. alternatively you can rename `HttpMock`.

File: src/app/services/signing/signing.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { Http } from '@angular/http';
+import { HttpMock } from './http.mock';
+import { SigningService } from './signing.service';
+
+describe('SigningService', () => {
+  let service: SigningService;
+  let http: HttpMock;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
```
Comment (lukaselmer) [inline src/app/services/signing/signing.service.spec.ts]: as already discussed: why so complicated? you can treat the `SigningService` as a normal object and construct it yourself, especially when you mock everything else away :) no need to use `TestBed` here.

**Review by lukaselmer — APPROVED** [review]
see comments 😉 otherwise it lgtm

**Pull request merged**